src/cljs/rems/table.cljs

```diff
@@ -63,18 +67,60 @@
   (reduce (fn [items {:keys [sort-column sort-order]}]
             (apply-sorting column-definitions sort-column sort-order items)) items initial-sort))
 
-(defn matches-filter [column-definitions col filter-value item]
-  (let [actual-value (str (column-filter-value column-definitions col item))]
-    (str/includes? (str/lower-case actual-value)
-                   (str/lower-case filter-value))))
-
-(defn matches-filters [column-definitions filters item]
-  (every? (fn [[col filter-value]] ()
-            (matches-filter column-definitions col filter-value item))
-          filters))
+(defn match-filter? [column-definitions filter-word item]
+  (some (fn [col]
+          (str/includes? (str/lower-case (column-filter-value column-definitions col item))
+                         (str/lower-case filter-word)))
+        (keys column-definitions)))
+
+(deftest test-match-filter
+ (is (true? (match-filter? {:id {:value :id}
+                            :description {:value :description}}
+                           "bar"
+                           {:id "foo"
+                            :description "bar"}))
+     "matches")
+ (is (not (match-filter? {:id {:value :id}
+                          :description {:value :its-a-trap
+                                        :filter-value :description}}
+                         "willnotmatch"
+                         {:id "foo"
+                          :its-a-trap "willnotmatch"
+                          :description "bar"}))
+     "doesn't match with value if filter-value is provided"))
+
+(defn match-filters? [column-definitions filters item]
+  (every? (fn [filter-word]
+            (match-filter? column-definitions filter-word item))
+          (str/split filters #"\s+")))
+
+(deftest test-match-filters
+ (is (true? (match-filters? {:id {:value :id}
+                             :description {:value :its-a-trap
+                                           :filter-value :description}}
+                            "foo bar"
+                            {:id "foo"
+                             :its-a-trap "willnotmatch"
+                             :description "bar"}))
+     "match with filter-value")
+ (is (not (match-filters? {:id {:value :id}
+                           :description {:value :its-a-trap
+                                         :filter-value :description}}
+                          "foo bar"
+                          {:id "foo"
+                           :its-a-trap "bar"
+                           :description "shouldnotmatch"}))
+     "doesn't match with value if filter-value is provided")
```

Comment:
This duplicates a lot of tests with `test-match-filter` so maybe just have the tests there and test some combination logic here.

Reply:
If you move the string splitting and lower-casing outside then this just tests `every?` that is not very useful.